ScnSoft-Education/LeetCodeTraining/DivideIntegers/DivideIntegersSolution.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace LeetCodeTraining.DivideIntegers
+{
+    //https://leetcode.com/problems/divide-two-integers/
+    public static class DivideIntegersSolution
+    {
+        public static int Divide(int dividend, int divisor)
+        {
+            if (dividend == 0)
+            {
+                return 0;
+            }
+
+            if (dividend == int.MinValue && divisor == -1)
+            {
+                return int.MaxValue;
+            }
+
+            int counter = 0;
+            bool negativeResult = (dividend < 0 && divisor > 0) ||
+                                  (dividend > 0 && divisor < 0);
+            dividend = dividend > 0 ? dividend -= dividend + dividend : dividend;
+            divisor = divisor < 0 ? divisor -= divisor + divisor : divisor;
+
+            while (dividend < 0)
+            {
+                counter++;
+                dividend += divisor;
+            }
+
+            if (dividend == 0)
+            {
+                return negativeResult ? counter -= counter + counter : counter;
+            }
+
+            return negativeResult ? counter -= counter + counter - 1 : counter - 1;
```

Comment:
return negativeResult ? -counter + 1 : counter - 1; ?

Reply:
Hello @shomzascnsoft
regarding this and the upper comments as well - I believe that according to problem's description we can not use "-number" construction because basically it means -1 * number, which is not allowed.
` ...divide two integers without using multiplication, division, and mod operator.`
Maybe it's the minor deviation from the problem's restrictions, and the algorithm itself was the problem, but I believe it's the only possible way of converting positive number to negative (and vise versa) in scope of this problem, aside from bit operations on int's 32th bit.

Reply:
Unary minus operator is not equalist to binary -1 * number operation:
https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1144-unary-minus-operator
![image](https://github.com/daniilnaumovich/ScnSoft-Education/assets/101803780/f93a386a-528d-4390-8f5d-b5d77bc11ad1)